src/net_processing.cpp

```diff
@@ -1308,9 +1309,37 @@ PeerManagerImpl::PeerManagerImpl(const CChainParams& chainparams, CConnman& conn
 }
 
 /**
- * Evict orphan txn pool entries based on a newly connected
- * block, remember the recently confirmed transactions, and delete tracked
- * announcements for them. Also save the time of the last tip update.
+ * If a transaction was removed from the mempool for a reason that entails it
+ * most likely will not re-enter, let the rebroadcast handler know.
+ */
+void PeerManagerImpl::TransactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRemovalReason reason, uint64_t mempool_sequence)
+{
+    if (!m_txrebroadcast) return;
+
+    switch(reason) {
+        case MemPoolRemovalReason::BLOCK:
+            // Although transactions removed for this reason will not be
+            // returned by this callback, include it here so the compiler
+            // can warn about missing cases in this switch statement.
+            // These transactions are handled by BlockConnected.
+            break;
```

Comment:
In 4165cb0606e4be476651899b80525f63f6e4465a [rebroadcast] Stop tracking rebroadcast attempts for certain transactions

Why not group `MemPoolRemovalReason::BLOCK` with the cases that we call `RemoveFromAttemptTracker()` for (while keeping the comment indicating that they don't go through this callback)?

Reply:
sure, doesn't seem like it matters either way? but done regardless